File: hw/astudillo/U3/HW18Factory/AbstractFactoryPython/ec/edu/espe/composite/abstractfactory/view/ClientApp.py
"""
Client Application

@author PC
"""
from ec.edu.espe.composite.abstractfactory.controller.GUIFactory import GUIFactory
from ec.edu.espe.composite.abstractfactory.model.Button import Button
from ec.edu.espe.composite.abstractfactory.model.Menu import Menu


class ClientApp:
    
    @staticmethod
    def main():
        a_factory = GUIFactory.get_factory()
        
        a_button = a_factory.create_button()
        a_button.caption = "Play"
        a_button.paint()
        
        a_menu = a_factory.create_menu()
        a_menu.caption = "File"
        a_menu.paint()


if __name__ == "__main__":
    ClientApp.main()

File: hw/andrade/u3/HW19AbstractFactory/HW19AbstractFactory/AbstractFactory/src/main/java/ec/edu/espe/abstractfactory/view/ClientApp.java

package ec.edu.espe.abstractfactory.view;

import ec.edu.espe.abstractfactory.controller.GUIFactory;
import ec.edu.espe.abstractfactory.model.Button;

/**
 *
 * @author Julio Andrade
 */
public class ClientApp {

    public static void main(String[] args){
        GUIFactory aFactory = GUIFactory.getFactory();
        Button aButton = aFactory.createButton();
        aButton.caption="Play";
        aButton.paint();
    }
}
